app/processing/statement/get-statement.js

```diff
@@ -0,0 +1,26 @@
+const { getDetails, getAddress, getDetailedFunding, getScheme, getDetailedPayments } = require('./components')
+const getCalculation = require('../calculation')
+const getPaymentRequest = require('../payment-request')
+const getSettlement = require('../settlement')
+
+const getStatement = async (settlementId) => {
+  const settlement = await getSettlement(settlementId)
+  const paymentRequestId = settlement.paymentRequestId
+  const paymentRequest = await getPaymentRequest(paymentRequestId)
+  const calculation = await getCalculation(paymentRequestId)
+  const details = await getDetails(calculation.sbi)
+  const address = await getAddress(calculation.sbi)
+  const detailedFunding = await getDetailedFunding(calculation.calculationId)
+  const scheme = await getScheme(paymentRequest)

```

Comment:
Can we make this explicit to what it takes in, i.e. `paymentRequest.agreementNumber`, `paymentRequest.frequency`, `paymentRequest.year`

Reply:
That is exactly what getScheme is doing : 

https://github.com/DEFRA/ffc-pay-statement-constructor/blob/PAY-2318-aggregate-map-and-build-statement-constructor-object/app/processing/statement/components/get-scheme.js

Reply:
My comment was to have the signature for `getScheme` be `getScheme(agreementNumber, frequency, year)` and then when we call it here `getScheme(paymentRequest.agreementNumber, paymentRequest.frequency, paymentRequest.year`

That way `getScheme` is acting on a known set of parameters which are better described and more predictable.

It also makes it consistent w/ every other func call above, e.g. `getAddress` signature is `getAddress(sbi)` rather than `getAddress(calculation)` and somewhere in the code `calculation.sbi` is used -- just makes it easier to test and more stable